src/nvim/option.c

```diff
@@ -2603,17 +2595,12 @@ did_set_string_option (
       xfree(*varp);
       *varp = p;
       if (varp == &p_enc) {
-        errmsg = mb_init();
-        redraw_titles();
+        // only encoding=utf-8 allowed
+        if (STRCMP(p_enc, "utf-8") != 0) {
```

Comment:
Should this check `p`? Or was `p_enc` already set to an invalid value by this point?